### PR TITLE
Dump flash in chunks

### PR DESCRIPTION
**Pull request by figgyc**
Tested on ak2i by jerbear. I also added a counter of how many chunks have been written.

### CODE REVIEW

File: source/nds_platform.cpp [hunk 1]
```diff
@@ -120,39 +120,58 @@ int InjectFIRM(flashcart_core::Flashcart* cart, bool isDevMode)
...
-	delete[] Flashrom;
-	fclose(FileOut);
 	fatUnmount("fat:/");
 
+    for (chunk=0; chunk < totalChunks; chunk++) {
+
+        DrawRectangle(TOP_SCREEN, FONT_WIDTH, SCREEN_HEIGHT-FONT_HEIGHT*2, SCREEN_WIDTH, FONT_HEIGHT, COLOR_BLACK);
+        DrawStringF(TOP_SCREEN, FONT_WIDTH, SCREEN_HEIGHT-FONT_HEIGHT*2, COLOR_WHITE, "Reading chunk 0x%x", chunk);
+
+	    u8 *Flashrom = new u8[chunkSize]; //Allocate a new array to store the flashrom we are about to retrieve from the flashcart
```
Comment (angelsl) [inline source/nds_platform.cpp]: Might want to hoist this out of the loop. No need to allocate and free the memory every iteration.

**Comment by angelsl** [comment]
Not sure if mounting and unmounting the FAT every iteration is needed either? I haven't seen any DLDI that deals with the partition being mounted, and in fact many of them plainly assume the SD is already initialised, so they're basically just direct block read/write commands, and flash reads shouldn't affect that.

File: source/nds_platform.cpp (hunk 1 above)
Comment (figgyc) [reply source/nds_platform.cpp]: fixed in dc751fa

**Comment by figgyc** [comment]
Yeah that was mainly just a failsafeish, I think @jason0597 may have said at one point that his r4i gold rev4 requires this but tbh I don't remember would probably be best to try doing it without and seeing if it still works

**Comment by jason0597** [comment]
the reason for the constant unmounting is because i feared some flashcarts would not like this when flashing/reading

if having fat mounted throughought the entire program does not affect any of the flashcart_core supported flashcarts during reading/flashing, then i will change it so that fat is mounted on boot

**Comment by jason0597** [comment]
you don't really need the chunk variable, you can just do
`for (u32 chunkOffset = 0; chunkOffset < Flash_size; chunkOffset += chunkSize)`
and get rid of lines 123, 125, 172, and also modify line 155

**Comment by jason0597** [comment]
also, could the current chunk be printed on the bottom screen, right on top of the progress bar?

**Comment by figgyc** [comment]
the progress bar routine clears the screen so it would disappear. I'll do the rest of the tweaks

**Pull request merged**